loader/base_loader.py

```diff
@@ -162,37 +163,70 @@ def get_gs_metadata_client(self, gcp_meta_cred):
         credentials = Credentials(token=None).from_authorized_user_file(gcp_meta_cred)
         return Client(project=self.google_project_id, credentials=credentials)
 
+    def handle_s3_client_error(self, err_code: str, bucket: str, key: str, attempt_refresh=True):
+        """
+        Will log warnings and consume the exception it was passed.  If the credentials get an
+        unauthorized/forbidden error, it will attempt to refresh metadata credentials (if they exist)
+        and try (only) once more.
+
+        :param bucket: Name of an S3 bucket
+        :param key: S3 file to upload.  err_code.g. 'output.txt' or 'data/output.txt'
+        :param attempt_refresh: Ensures attempting to refresh the metadata credentials happens only once per file.
+        :return: Returns an empty dict or a head response containing a dictionary of metadata values.
```

Comment:
I would put the typical expected case first, followed by what is returned in the case of an error. Something like:
```
Returns a head response containing a dictionary of metadata values, or an empty dict in the case of an error.
```